### PR TITLE
[Components] Memberspot.io #7834

### PR DESCRIPTION
**Pull request by lcaresia**
### <samp>🤖 Generated by Copilot at 58b597c</samp>

This pull request adds a new app component and two action components for the Memberspot app, which is a platform for creating and selling online courses and memberships. The pull request also updates the `pnpm-lock.yaml` file to ensure consistent dependency versions. The pull request deletes some unused files from the previous Memberspot app component.

### <samp>🤖 Generated by Copilot at 58b597c</samp>

> _To manage Memberspot access_
> _We added some components, no less_
> _They use `axios`_
> _And `memberspot.app`_
> _And update the `pnpm-lock` mess_

### <samp>🤖 Generated by Copilot at 58b597c</samp>

*  Add new action components for the Memberspot app to grant and delete access for users by email and offer ID ([link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-79ee4ab33001c46e083d864cf7ff4a9085e1cbf15efb5bd95a038b0940d1834eR1-R39), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-8be7cc1fad43625017c650010b0459ec5fcfd26c7f25f9aff4feb55e3f80f640R1-R50))
*  Add new app component for the Memberspot app to define authentication, prop definitions and methods for interacting with the Memberspot API ([link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-89b961008d6f0afd73a6e56940f5bff1b5fbfb6e3aa5a2fc9b39877dfd32a304R1-R60))
*  Update package.json file for the Memberspot app component to include the @pipedream/platform dependency, change the main entry point and bump the version ([link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-2d8b1bd5e351f30b12910a4ae224450cd2bc63d38b83539fb113f94576ea293bL3-R18))
*  Delete unused files for the Memberspot app component ([link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-3aa047f85cc7744cd6fbdb906ca34d1782fc745d1aee649c6883764de2ef58ae), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-c2b41b4ee9d1f8096840d66a8263f551473812ef91d4a77e63950399712a00e2))
*  Add or update entries for various app components to the pnpm-lock.yaml file to lock the dependency versions ([link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR190-R191), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR363-R364), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1130-R1131), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1198-R1199), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1361-R1362), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1744-R1747), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2503-R2504), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2544-R2545), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2643-R2644), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL2918-R2942), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3370-R3371), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3766-R3767), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3831-R3832), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR3933-R3934), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4023-R4024), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4076-R4077), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4107-R4108), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4184-R4185), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4221-R4222), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4392-R4393), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4713-R4714), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4947-R4948), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4988-R4989), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5238-R5239), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5478-R5479), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5499-R5500), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5713-R5714), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR5815-R5816), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8932-R9007), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL15652-R15712), [link](https://github.com/PipedreamHQ/pipedream/pull/7932/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL15668-R15730))

### CODE REVIEW

File: components/memberspot/actions/delete-access/delete-access.mjs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import app from "../../memberspot.app.mjs";
+
+export default {
+  name: "Delete Access",
+  version: "0.0.1",
+  key: "memberspot-delete-access",
+  description: "Delete access. [See the documentation](https://api.memberspot.de/api/#/users/UsersApiController_setOfferState)",
```
Comment (alysonturing) [inline components/memberspot/actions/delete-access/delete-access.mjs:7]: Could you elaborate more on the description?

File: components/memberspot/actions/grant-access/grant-access.mjs [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import app from "../../memberspot.app.mjs";
+
+export default {
+  name: "Grant Access",
+  version: "0.0.1",
+  key: "memberspot-grant-access",
+  description: "Grant access. [See the documentation](https://api.memberspot.de/api/#/users/UsersApiController_grantUserOfferByMail)",
```
Comment (alysonturing) [inline components/memberspot/actions/grant-access/grant-access.mjs:7]: Could you elaborate more on the description?

**Review by alysonturing — CHANGES_REQUESTED** [review]
Could you elaborate more on the descriptions?

**Review by michelle0927 — APPROVED** [review]
Approved!

**Pull request merged**